doc/api/stream.md

```diff
@@ -1590,6 +1596,18 @@ async function run() {
 run().catch(console.error);
 ```
 
+On completion or error `stream.pipeline()` will call `stream.destroy(err)` on
+all provided stream except for `Readable` streams which have emitted `'end'`
+or `'close'` and `Writable` streams which have emitted `'finish'` or `'close'`.
+Unless `autoDestroy` is enabled these streams will not be automatically cleaned
+up.
+
+If a `Writable` stream emits `'close'` before `writable.writableFinished'`,
+`stream.pipeline()` will error with `ERR_STREAM_PREMATURE_CLOSE`.
+
+If a `Readable` stream emits `'close'` before `readable.push(null)`,
```

Comment:
We might want to add a `Readable.readableFlushed` property to inspect this specific state.